File: backend/schemas/story.py
from pydantic import BaseModel
from typing import List


class VisualPrompt(BaseModel):
    """Schema for visual prompt generation."""
    description: str
    camera_angle: str
    mood: str


class Scene(BaseModel):
    """Schema for a single story scene."""
    id: int
    text_content: str
    visual_prompts: VisualPrompt
    estimated_duration: int


class Story(BaseModel):
    """Schema for a complete viral story."""
    title: str
    topic: str
    target_audience: str
    scenes: List[Scene]
    clickbait_score: int


class ViralStory(BaseModel):
    """Schema for a viral story with AI thinking trace."""
    title: str
    topic: str
    target_audience: str
    scenes: List[Scene]
    clickbait_score: int
    thinking_trace: str  # AI's internal reasoning about why this story will go viral

File: frontend/src/types/story.ts
/**
 * TypeScript interfaces matching the Pydantic models in backend/schemas/story.py
 */

export interface VisualPrompt {
  description: string;
  camera_angle: string;
  mood: string;
}

export interface Scene {
  id: number;
  text_content: string;
  visual_prompts: VisualPrompt;
  estimated_duration: number;
}

export interface Story {
  title: string;
  topic: string;
  target_audience: string;
  scenes: Scene[];
  clickbait_score: number;
}

export interface ViralStory {
  title: string;
  topic: string;
  target_audience: string;
  scenes: Scene[];
  clickbait_score: number;
  thinking_trace: string; // AI's internal reasoning about why this story will go viral
}

